source/blog/2018-06-21-ember-3-2-released.md

```diff
@@ -0,0 +1,350 @@
+
+This is, to put it quite simply, totally awesome. It means that this code:
+
+```javascript
+assert.equal(this.element.querySelector('.title').textContent.trim(), 'Hello World!');
+```
+
+becomes this:
+
+```javascript
+assert.dom('.title').hasText('Hello World!');
+```
+
+See what I mean? Totally awesome. <3
+
+##### Experiments with more efficient transpilation (2 of 3)
+
+Until now, addons were responsible for compiling their own JS/HBS/CSS and returning AMD/CSS. Now they return the raw code, and the app uses its own processors (babel, htmlbars) to compile it. This is required to do proper tree-shaking and code-splitting. Delayed transpilation [(#7501)](https://github.com/ember-cli/ember-cli/pull/7501) and all-at-once addon optimization after compilation [(#7650)](https://github.com/ember-cli/ember-cli/pull/7650) have been added. Additionally, more comprehensive methods to detect if ember-cli is being run within CI or not have also been added [(#7637)](https://github.com/ember-cli/ember-cli/pull/7637) - see [https://github.com/watson/ci-info/](https://github.com/watson/ci-info/).
+
+##### Module Unification (new file layout) Continues (3 of 3)
+
+You can now generate an addon using the Module Unification layout [(#7490)](https://github.com/ember-cli/ember-cli/pull/7490)! Use the command `MODULE_UNIFICATION=true ember addon my-addon` to try it out [(#7658)](https://github.com/ember-cli/ember-cli/pull/7658). We also improved the logic to support addons that use Module Unification [(#7660)](https://github.com/ember-cli/ember-cli/pull/7660), added the blueprint for a dummy app to addons that use Module Unification [(#7667)](https://github.com/ember-cli/ember-cli/pull/7667), and updated the version of Ember used in Module Unification [(#7678)](https://github.com/ember-cli/ember-cli/pull/7678).
+
+#### Deprecations (1)
+
+##### ember-cli-babel 5 (1 of 1)
+
+This release of Ember CLI [deprecates `ember-cli-babel` 5.x](https://github.com/ember-cli/ember-cli/pull/7676). Babel 6 support has been out for a long time now and works quite well. Babel 5 support is deprecated and is expected to be dropped soon.
+
+For more details on the changes in Ember CLI 3.2.0-beta.1 and detailed upgrade
+instructions, please review the [Ember CLI 3.2.0-beta.1 release page](https://github.com/ember-cli/ember-cli/releases/tag/v3.2.0-beta.1).
+
+Thank you to [@GavinJoyce](https://github.com/GavinJoyce), [@Turbo87](https://github.com/Turbo87), [@cibernox](https://github.com/cibernox), [@iezer](https://github.com/iezer), [@kellyselden](https://github.com/kellyselden), [@raytiley](https://github.com/raytiley), [@t-sauer](https://github.com/t-sauer), and [@thetimothyp](https://github.com/thetimothyp)
+for your incredible work on ember-cli!
+
+For more details on the changes in Ember CLI 3.2 and detailed upgrade
+instructions, please review the [Ember CLI  3.2.0 release page](https://github.com/ember-cli/ember-cli/releases/tag/v3.2.0).
+
+### Changes in Ember CLI 3.3
```

Comment:
```
Ember CLI 3.3 will include a lot of internal changes to enable the new
[Packager](https://github.com/ember-cli/rfcs/blob/master/active/0051-packaging.md)
RFC. Work on [Module Unification](https://github.com/emberjs/rfcs/blob/master/text/0143-module-unification.md)
is also still in progress and for Ember CLI 3.3 we have changed the way feature flags
for Ember CLI work. You can read about the details in the
[docs section](https://github.com/ember-cli/ember-cli/blob/master/docs/experiments.md)
of the repository.

There are three other notable changes:

#### Node.js support

Node 4 is at the end of its lifecycle and that means Ember CLI is dropping support
for it, as outlined in the [blog post](https://www.emberjs.com/blog/2016/09/07/ember-node-lts-support.html)
about our Node.js support policy. At the same time we have added official support for
the new Node 10 release.

#### `ember-cli-shims`

With the deprecation of Babel 5, and Babel 6 supporting imports from
[`@ember/...` paths](https://github.com/ember-cli/ember-rfc176-data/) directly there
is no more need for the `ember-cli-shims` dependency.
PR [#7774](https://github.com/ember-cli/ember-cli/pull/7774) has been merged which removes
the dependency from the default app blueprint.

#### jQuery in addons

Ember.js itself no longer needs jQuery, and most addons don't need it either. We have
changed the default addon blueprint to disable jQuery by default to make sure that addon
authors test their addons without jQuery being present. If your addon is using jQuery
you can of course keep using it, but new addons should try to avoid it.
```

/cc @kategengler